File: server/src/main/java/com/ps/server/dto/PromotionDTO.java
package com.ps.server.dto;

import com.ps.server.Logic.Pieces.Piece;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PromotionDTO {
    private Long playerId;
    private Piece.PieceType pieceType;
}

File: client/src/app/DTO/promotionDTO.ts
import {PieceType} from "../_models/piece";

/** @ignore*/
export class PromotionDTO {
    /** */
    public playerId: number;
    /** */
    public pieceType: PieceType;

    /** @ignore*/
    constructor(playerId: number, pieceType: PieceType) {
        this.playerId = playerId;
        this.pieceType = pieceType;
    }
}